Core tests for UDP IPv4 and IPv6

Sanjhana-Jayagopal opened the pull request:
#### What does this PR do?
Fixes #91 
Adds test scenarios for the core switch using UDP IPv4 and IPv6 packets.
#### Do you have any concerns with this PR?
TCP IPv4 and IPv6 does not work with this branch.
#### How can the reviewer verify this PR?
Run CI.
#### Any background context you want to provide? No
#### Screenshots or logs (if appropriate) N/A
#### Questions:
- Have you connected this PR to the issue it resolves? Yes
- Does the documentation need an update? No
- Does this add new dependencies? No
- Have you added unit or functional tests for this PR?

spisarski committed:
Branch for testing changes to CI jobs

Commit:
core.p4 changes

Commit:
core.p4 changes

Commit:
Moved data_inspection to egress processing

Commit:
Moved data_inspection to egress processing

spisarski committed:
Fixes to playbooks from warnings logged by the ansible-lint python package

spisarski committed:
fixed multi-line command

spisarski committed:
removed state in pip task

spisarski committed:
Fixed final ansible-lint warning

spisarski committed:
Removed become: yes as this will break the plays when running with run_mode=remote. Moving back to using sudo on the command calls.

spisarski committed:
Fixed newline being added to send_packets.py when an INT header is required to be mocked up which could become an issue.

Commit:
Core tests for UDP IPv4/IPv6

Commit:
Core tests for UDP IPv4/IPv6

Commit:
Merge remote-tracking branch 'origin/core-tests' into core-tests

Commit:
Core tests for UDP IPv4/IPv6

Commit:
Merge remote-tracking branch 'origin/core-tests' into core-tests

Commit:
Comments

spisarski commented on bin/write_to_clone.py:
```diff
@@ -0,0 +1,91 @@
+    return parser.parse_args()
+
+
+def read_yaml_file(config_file_path):
```
Remove this function as it is not being used

Commit:
Updates to write_clone_entry and clear_clone-entry

spisarski commented on playbooks/scenarios/test_cases/write_clone_entry.yml:
```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2019 Cable Television Laboratories, Inc.
```
Please remove clear_clone_entry.yml and just use write_clone_entry and set the var.clone from core/data-inspection.yml

spisarski commented on playbooks/scenarios/test_cases/clear_clone_entry.yml:
```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2019 Cable Television Laboratories, Inc.
```
DELETE ME - This is redundant with write_clone_entry.yml minus line #23 which can be sent in from the caller

spisarski reviewed (changes requested):
Please see my comments. Logic basically looks fine, just gotta remove some unused and redundant stuff.

Commit:
Updates to write_to_clone

Commit:
Updated receiver interfaces for core test cases

spisarski commented:
I will be making a few fixes...

spisarski commented:
Closing PR to perform more work as noted on issue #91 
The biggest issue is that the INT packets are getting sent out to port 2 and stripped packets out port 3. I may need to return the re-circulation logic back.

spisarski commented:
Fixing:

1. hdr.ipv4.totalLen
2. activating all tests for TCP/UDP and IPv4/IPv6
3. to fix shim length in mocked packets
4. create device for cloned packets
5. minor refactoring
6. speeding tests by simplification
7. improve test packet receiver parsing

spisarski commented:
Packets seemed to be cloning properly. May have incorrectly determined the port issue mentioned above potentially due to the hdr.ipv4.totalLen

spisarski committed:
fix to p4 hdr.ipv4.totalLen
Activating all IPv4/IPv6 TCP/UDP scenarios for core
Fixed mocked INT packets shim length value
Minor refactoring at base test playbooks with minor key changes on client playbooks
Improved test times by removing non-critical packet tests
Small improvement parsing test sniffer log file for packet counts

spisarski commented:
Fixes worked while running in development environment with run_mode=local for all tests.

spisarski committed:
bug fix, wrong host

spisarski committed:
fix receiver on egress port scenario
